src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -580,21 +583,27 @@ public String getWriteableName() {
         return NAME;
     }
 
-    private static void validateSingleQueryType(Integer k, Float distance, Float score) {
+    private static VectorQueryType validateSingleQueryType(Integer k, Float distance, Float score) {
         int countSetFields = 0;
```

Comment:
Can we avoid all together customeSetFields and fall early based on if all of the parameters are null or zero. Which is going to simply all 3 ifs we are putting together

Reply:
We have to verify these three parameters together because we need check only exactly one of them can be pass in.